### ISSUE TITLE
这个CNN_heartbeat_for_pubDataset.py能用你提供heartbeat data训练？

### ISSUE THREAD

**Issue by liweikang822** [HYChang2022/Vital-Sign-Estimation-with-Deep-Learning-Aided-Weighted-Scheme-Using-FMCW-Radar#1]

**Comment by HYChang2022** [comment]
Yes, it is applicable after the following steps are achieved.

1. First copy "rawData_* .bin" files from data directories and run the "DAWS_algorithm_code\data preprocessing\bin to mat\bin_to_mat.m" code to acquire "radarSignal_* .mat" files. Note that the input and output settings of files in "bin_to_mat.m" might require manual set.
2. Unzip "DAWS_algorithm_code\data\data_beat_test\heartbeat\ground_truth.zip" and "DAWS_algorithm_code\data\data_beat_train\heartbeat\ground_truth.zip". Note that there are four zipped files in total.
3. Run "DAWS_algorithm_code\RPM_generator.m" after properly modify the following setting in lins 8--9:

```
training_or_testing = 0;     % 0: training data,  1: testing data
breathing_or_heartbeat = 1;  % 0: breathing data, 1: heartbeat data
```

4. Now you are ready to train DAWS with "DAWS_algorithm_code\CNN_heartbeat_for_pubDataset.py"

I hope that answers your question. If you need any further assistance, feel free to ask.

**Comment by liweikang822** [comment]
thank yoThank you! I have finished training your model. The performance is good, but I noticed that the testing and training data you provided were obtained from three individuals, with heart rates ranging from 70 to 90. Do you have any heart rate radar data for the range of 50-70 and 90-110? How does the model perform on this range, in terms of loss?u i have trained you model，

**Comment by liweikang822** [comment]
Thank you! I have finished training your model. The performance is good, but I noticed that the testing and training data you provided were obtained from three individuals, with heart rates ranging from 70 to 90. Do you have any heart rate radar data for the range of 50-70 and 90-110? How does the model perform on this range, in terms of loss?

**Comment by HYChang2022** [comment]
The heart rate radar data in the range of 50-70 and 90-110 was addressed in the study "Radar recorded child vital sign public dataset and deep learning-based age group classification framework for vehicular application" by Yoo et al. However, due to limited experiment duration, a CNN model couldn't be adequately trained within these ranges. Consequently, the model's performance within these specific ranges remains untested. It's uncertain whether the observed performance issues are due to the range itself or the insufficient data available for training.

I hope that answers your question. If you need any further assistance, feel free to ask.
